backend/internal/userschema/service/userschemaservice.go

```diff
@@ -0,0 +1,505 @@
+	CreateUserSchema(request model.CreateUserSchemaRequest) (*model.UserSchema, *serviceerror.ServiceError)
+	GetUserSchema(schemaID string) (*model.UserSchema, *serviceerror.ServiceError)
+	UpdateUserSchema(schemaID string, request model.UpdateUserSchemaRequest) (
+		*model.UserSchema, *serviceerror.ServiceError)
+	DeleteUserSchema(schemaID string) *serviceerror.ServiceError
+}
+
+// UserSchemaService is the default implementation of the UserSchemaServiceInterface.
+type UserSchemaService struct{}
+
+// GetUserSchemaService creates a new instance of UserSchemaService.
+func GetUserSchemaService() UserSchemaServiceInterface {
+	return &UserSchemaService{}
+}
+
+// GetUserSchemaList lists the user schemas with pagination.
+func (us *UserSchemaService) GetUserSchemaList(limit, offset int) (
+	*model.UserSchemaListResponse, *serviceerror.ServiceError) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, userSchemaLoggerComponentName))
+
+	if err := validatePaginationParams(limit, offset); err != nil {
+		return nil, err
+	}
+
+	totalCount, err := store.GetUserSchemaListCount()
+	if err != nil {
+		return nil, logErrorAndReturnServerError(logger, "Failed to get user schema list count", err)
+	}
+
+	userSchemas, err := store.GetUserSchemaList(limit, offset)
+	if err != nil {
+		return nil, logErrorAndReturnServerError(logger, "Failed to get user schema list", err)
+	}
+
+	response := &model.UserSchemaListResponse{
+		TotalResults: totalCount,
+		StartIndex:   offset + 1,
+		Count:        len(userSchemas),
+		Schemas:      userSchemas,
+		Links:        buildPaginationLinks(limit, offset, totalCount),
+	}
+
+	return response, nil
+}
+
+// CreateUserSchema creates a new user schema.
+func (us *UserSchemaService) CreateUserSchema(request model.CreateUserSchemaRequest) (
+	*model.UserSchema, *serviceerror.ServiceError) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, userSchemaLoggerComponentName))
+
+	if request.Name == "" {
+		return nil, &constants.ErrorInvalidUserSchemaRequest
+	}
+
+	if len(request.Schema) == 0 {
+		return nil, &constants.ErrorInvalidUserSchemaRequest
+	}
+
+	if err := validateJSONSchema(request.Schema); err != nil {
+		return nil, &constants.ErrorInvalidUserSchemaRequest
+	}
+
+	_, err := store.GetUserSchemaByName(request.Name)
+	if err == nil {
+		return nil, &constants.ErrorUserSchemaNameConflict
+	} else if !errors.Is(err, constants.ErrUserSchemaNotFound) {
+		return nil, logErrorAndReturnServerError(logger, "Failed to check existing user schema", err)
+	}
+
+	schemaID := utils.GenerateUUID()
+
+	userSchema := model.UserSchema{
+		ID:     schemaID,
+		Name:   request.Name,
+		Schema: request.Schema,
+	}
+
+	if err := store.CreateUserSchema(userSchema); err != nil {
+		return nil, logErrorAndReturnServerError(logger, "Failed to create user schema", err)
+	}
+
+	logger.Info("User schema created successfully", log.String("schemaID", schemaID), log.String("name", request.Name))
```

Comment:
Do we need a info log here?

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/2359bb281523048120acb1f8c4d2b4ce08b10f5f..d0b7099012c7c17094c1feb4e8baa826f84603be